### PR TITLE
Added scroll down pagination for data-navigation

### PR DESCRIPTION
**Pull request by wachterjohannes**
__Tasks:__

- [ ] gather feedback for my changes

__Informations:__

| Q             | A
| ------------- | ---
| Tests pass?   | yes
| Fixed tickets | none
| BC Breaks     | none
| Doc           | none

### CODE REVIEW

File: husky_components/data-navigation/main.js [hunk 1]
```diff
@@ -237,6 +306,18 @@ define([
             this.sandbox.on(CLEAR_CACHE.call(this), function() {
                 this.cache.deleteAll();
             }.bind(this));
+
+            this.sandbox.on('husky.search.data-navigation', function(searchTerm) {
+                this.searchTerm = searchTerm;
```
Comment (danrot) [inline husky_components/data-navigation/main.js]: Is this really necessary? Looks like a source of bugs to me. Can't we just read this value from the text field in the `getCurrentUrl` field?

Comment (wachterjohannes) [inline husky_components/data-navigation/main.js]: reading from a input which is created by a component is dirty, then we should create an event and return this over a callback! but then `getCurrentUrl` has to call this event everytimes ...

Comment (danrot) [inline husky_components/data-navigation/main.js]: Or can we maybe pass it to this method as a parameter?

File: husky_extensions/infinite-scroll.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ */
+(function() {
+
+    'use strict';
+
+    define(function() {
+
+        var scrollHandler = function(event, sandbox, padding, callback) {
+            var $e = sandbox.dom.find(event.currentTarget),
```
Comment (danrot) [inline husky_extensions/infinite-scroll.js]: Do not use abbreviations, call it `$element`, `$eventTarget` or whatever.

File: husky_extensions/infinite-scroll.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                    sandbox.dom.find('div.iscroll-inner', $e)
+                ),
+                borderTopWidth = parseInt(sandbox.dom.css($e, 'borderTopWidth')),
+                borderTopWidthInt = isNaN(borderTopWidth) ? 0 : borderTopWidth,
+                iContainerTop = parseInt(sandbox.dom.css($e, 'paddingTop')) + borderTopWidthInt,
+                iTopHeight = sandbox.dom.offset($e).top,
+                innerTop = $inner.length ? sandbox.dom.offset($inner).top : 0,
+                iTotalHeight = Math.ceil(iTopHeight - innerTop + $e.height() + iContainerTop);
+
+            if (sandbox.dom.data($e, 'enabled') === 'on' && iTotalHeight + (isNaN(padding) ? 0 : padding) >= $inner.outerHeight()) {
```
Comment (danrot) [inline husky_extensions/infinite-scroll.js]: Is this line short enough?

File: husky_extensions/infinite-scroll.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                sandbox.dom.data($e, 'enabled', 'off');
+                var result = callback();
+
+                if (!!result && !!result.then) {
+                    result.then(function() {
+                        sandbox.dom.data($e, 'enabled', 'on');
```
Comment (danrot) [inline husky_extensions/infinite-scroll.js]: What's this enabled for?

File: husky_components/data-navigation/main.js (hunk 1 above)
Comment (wachterjohannes) [inline husky_components/data-navigation/main.js]: i need it to load for example the second page of search results ....

File: husky_extensions/infinite-scroll.js (hunk 4 above)
Comment (wachterjohannes) [inline husky_extensions/infinite-scroll.js]: i will change the name of this property to blocked!

File: husky_extensions/infinite-scroll.js (hunk 2 above)
Comment (wachterjohannes) [inline husky_extensions/infinite-scroll.js]: +1

File: husky_extensions/infinite-scroll.js (hunk 3 above)
Comment (wachterjohannes) [inline husky_extensions/infinite-scroll.js]: no (=

**Comment by wachterjohannes** [comment]
@danrot everything fixed!

**Pull request merged**